### PR TITLE
ath79: add support for onion omega

### PR DESCRIPTION
**Pull request by AiyionPrime**
this partly reverts fc553c7e4c8eea898aaa2086574b7e8737f6d26c

### CODE REVIEW

File: target/linux/ath79/dts/ar9331_onion_omega.dts [hunk 1]
```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0
```
Comment (karlp) [inline target/linux/ath79/dts/ar9331_onion_omega.dts]: you may consider gpl2 or later, or even dual licensed, see other files for examples, but strict gpl2 _only_ is perhaps (or not!) not whatyou intended?

File: target/linux/ath79/dts/ar9331_onion_omega.dts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	status = "okay";
+
+	compatible = "syscon", "simple-mfd";
+};
+
+&eth1 {
+	status = "okay";
+
+	mtd-mac-address = <&uboot 0x01fc00>;
+	mac-address-increment = <(-1)>;
```
Comment (karlp) [inline target/linux/ath79/dts/ar9331_onion_omega.dts:75]: mtd mac will _work_ but openwrt master just recently moved to the "nvmem" style.  This _should_ be fairly straightforward to figure out from the recent commits...

File: target/linux/ath79/dts/ar9331_onion_omega.dts (hunk 1 above)
Comment (AiyionPrime) [reply target/linux/ath79/dts/ar9331_onion_omega.dts]: I took the license from the original ar71xx commit; as I was not sure whether work based on that could be re-licensed without the original authors consent?

Comment (karlp) [reply target/linux/ath79/dts/ar9331_onion_omega.dts]: nah, dts is completley new file, and it's based on the hardware itself.  copyright on the old mach file is irrelevant

File: target/linux/ath79/dts/ar9331_onion_omega.dts (hunk 2 above)
Comment (AiyionPrime) [reply target/linux/ath79/dts/ar9331_onion_omega.dts:75]: Got it resolved, I think. Compiles and runs both with a sysupgrade and a factory flash. Thanks!

**Comment by AiyionPrime** [comment]
https://patchwork.ozlabs.org/project/openwrt/patch/20210811165437.2280760-1-git@aiyionpri.me/

**Pull request closed without merging**